packages/backend-common/src/tokens/ServerTokenManager.ts

```diff
@@ -155,7 +155,7 @@ export class ServerTokenManager implements TokenManager {
         .setExpirationTime(
           DateTime.now().plus(TOKEN_EXPIRY_AFTER).toUnixInteger(),
         )
-        .sign(this.signingKey);
+        .sign(new TextEncoder().encode(this.signingKey.toString()));
```

Comment:
The signing key is already binary data at this point, it's decoded from base64 data so it would be up to that underlying data to already be encoded correctly afaik.

Reply:
Please note that a lot of us has generated our keys when it has a different algorithm so most of us as stored keys that are not url encoded. It would not be feasible to make us rotate all our keys to make this work.

Would it be better then if I encode the stored keys when we read it from the conf file?

Reply:
added changes in new commit

Reply:
Could you back up a bit and describe the actual issue you're seeing? This new solution seems very unrelated to the original change, text encoding and URL encoding are very different things.

Reply:
Actually, you are right. I added new changes.
In this implementation, the secret is now encoded in the constructor to produce the signingKey.

Reply:
My only concern now is what to do with the generateKeys() dev method. Should I change `this.signingKey = this.verificationKeys[0]` to `this.signingKey = TEXT_ENCODER.encode(secret)` or `this.signingKey = TEXT_ENCODER.encode(jwk.k)`??

Reply:
Sorry time to back up further 😅, the fix here is not really related to #18622. That issue is about invalid `alg` header values in the JWT due to a token mixup. This is making changes to how the backend secrets are encoded. What is the issue that you are seeing that you're aiming to fix?

Reply:
Sorry I wasn't able to work on much on this since I was busy, I had more time to read more on the documentation for signing tokens and updated my description.

For that thread, I was getting that same error that they are complaining about which is `invalid alg`. I debugged the code and found that the signingkeys and verification keys are being decoded instead of encoded, and after fixing it, the `invalid alg` error is not showing anymore